test/e2e/tekton/taskrun.yaml

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 The Tekton Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: kind
+spec:
+  taskSpec:
+    params:
+      - name: repo
+        description: git repo url to clone
+        type: string
+        default: https://github.com/tektoncd/results
+      - name: ref
+        description: git repo ref to checkout
+        type: string
+        default: main
+    # TODO: Refactor this to a Pipeline to make the kind task reusable.
+    steps:
+      - image: docker.io/alpine/git
+        name: checkout
+        script: |
+          git clone $(params.repo) .
+          git checkout $(params.ref)
+      # TODO: Move image to tekton-dogfooding (or other similar location)
+      - image: gcr.io/wlynch-test/kind-runner
+        name: kind
+        securityContext:
+          privileged: true
```

Comment:
Is it required for `kind` itself ? we may need to run as root, but I don't think we need the extra capabilities `privileged` adds, maybe ?

Reply:
Looks like we can remove this! 🎉